@here/harp-map-theme/resources/berlin_tilezen_base.json

```diff
@@ -86,19 +91,125 @@
     "styles": {
         "tilezen": [
             {
-                "description": "highway-label",
-                "when": "$layer == 'roads' && ((kind == 'highway') && ((kind_detail == 'motorway')))",
-                "technique": "text",
+                "description": "highway-roadshield-2",
+                "when": "$layer == 'roads' && (kind == 'highway') && (kind_detail == 'motorway') && has(ref) && (length(ref) <= 2)",
+                "technique": "line-marker",
                 "attr": {
-                    "color": "#000000",
-                    "backgroundColor": "#FFFFFF",
-                    "backgroundOpacity": 0.5,
-                    "size": 16,
-                    "priority": 25,
-                    "fadeNear": 0.8,
-                    "fadeFar": 0.9
+                    "style": "smallSign",
+                    "label": "ref",
+                    "size": 10,
+                    "imageTexture": "default-2",
+                    "iconScale": 1,
+                    "priority": 64,
+                    "minDistance": 200,
+                    "vAlignment": "Center",
+                    "hAlignment": "Center",
+                    "textIsOptional": false,
+                    "iconIsOptional": false,
+                    "textMayOverlap": false,
+                    "iconMayOverlap": false,
+                    "iconReserveSpace": true,
+                    "renderTextDuringMovements": true,
+                    "color": "#000000"
                 },
-                "renderOrder": 10
+                "renderOrder": 14.2
+            },
+            {
+                "description": "highway-roadshield-3, level > 7",
+                "when": "$layer == 'roads' && (kind == 'highway') && (kind_detail == 'motorway') && has(ref) && (length(ref) == 3) && ($level > 7)",
+                "technique": "line-marker",
+                "attr": {
+                    "style": "smallSign",
+                    "label": "ref",
+                    "size": 10,
+                    "imageTexture": "default-3",
+                    "iconScale": 1,
+                    "priority": 63,
+                    "minDistance": 200,
+                    "vAlignment": "Center",
+                    "hAlignment": "Center",
+                    "textIsOptional": false,
+                    "iconIsOptional": false,
+                    "textMayOverlap": false,
+                    "iconMayOverlap": false,
+                    "iconReserveSpace": true,
+                    "renderTextDuringMovements": true,
+                    "color": "#000000"
+                },
+                "renderOrder": 14.3
```

Comment:
@Astrak, @
`renderOrder` is not needed for techniques drawn in "2d" space

Reply:
Ok, then I will remove it.